src/Cms/Page.php

```diff
@@ -769,18 +769,32 @@ public function isUnlisted(): bool
 	 */
 	public function isVerified(string|null $token = null): bool
 	{
+		// if a valid token exists, the page is definitely verified
+		if ($this->isVerifiedByToken($token) === true) {
+			return true;
+		}
+
+		// public pages don't need extra verification
 		if (
 			$this->isPublished() === true &&
 			$this->parents()->findBy('status', 'draft') === null
 		) {
 			return true;
 		}
 
-		if ($token === null) {
-			return false;
+		return false;
+	}
+
+	/**
+	 * Validates the given token
+	 */
+	public function isVerifiedByToken(string|null $token): bool
+	{
+		if ($token !== null && $this->token() === $token) {
+			return true;
 		}
 
-		return $this->token() === $token;
+		return false;
```

Comment:
`Question`: What's the purpose of having `::isVerified($token)` as well as `::isVerifiedByToken()`? If both are essential separately, can we find more distinct names that explain their individual purpose?

Reply:
How about rather `::isPreviewable($token)` and `::isValidPreviewToken()`?